snapcraft/repo.py

```diff
@@ -58,15 +58,19 @@
 
 
 def install_build_packages(packages):
+    seen = set()
+    seen_add = seen.add
+    unique_packages = [x for x in packages if not (x in seen or seen_add(x))]
```

Comment:
it seems my comment from before did not stick :hushed: 

Is this a way to get only one item? If so, instead of

```python
seen = set()
seen_add = seen.add
unique_packages = [x for x in packages if not (x in seen or seen_add(x))]
```

do 

```python
unique_packages = set(packages)
```

Or bellow in

```python
for pkg in unique_packages
```

Change that to

```python
for pkg in set(packages)
```